samples/nrf9160/modem_shell/src/location/location_shell.c

```diff
@@ -0,0 +1,401 @@
+	"                      (default: 0 = single position)\n"
+	"  --gnss_accuracy,    Used GNSS accuracy: 'low' or 'normal'\n"
+	"  --gnss_timeout,     GNSS timeout in seconds\n"
+	"  --cellular_timeout, Cellular timeout in seconds\n"
+	"  --cellular_service, Used cellular positioning service:\n"
+	"                      'any' (default), 'nrf', 'skyhook', 'here' or 'polte'\n"
+	"  --wifi_timeout,     WiFi timeout in seconds\n"
+	"  --wifi_service,     Used WiFi positioning service:\n"
+	"                      'any' (default), 'nrf', 'skyhook' or 'here'\n";
+
+/******************************************************************************/
+
+/* Following are not having short options */
+enum {  LOCATION_SHELL_OPT_INTERVAL         = 1001,
+	LOCATION_SHELL_OPT_GNSS_ACCURACY    = 1002,
+	LOCATION_SHELL_OPT_GNSS_TIMEOUT     = 1003,
+	LOCATION_SHELL_OPT_CELLULAR_TIMEOUT = 1004,
+	LOCATION_SHELL_OPT_CELLULAR_SERVICE = 1005,
+	LOCATION_SHELL_OPT_WIFI_TIMEOUT     = 1006,
+	LOCATION_SHELL_OPT_WIFI_SERVICE     = 1007,
+};
+
+/* Specifying the expected options */
+static struct option long_options[] = {
+	{ "method", required_argument, 0, 'm' },
+	{ "interval", required_argument, 0, LOCATION_SHELL_OPT_INTERVAL },
+	{ "gnss_accuracy", required_argument, 0, LOCATION_SHELL_OPT_GNSS_ACCURACY },
+	{ "gnss_timeout", required_argument, 0, LOCATION_SHELL_OPT_GNSS_TIMEOUT },
+	{ "cellular_timeout", required_argument, 0, LOCATION_SHELL_OPT_CELLULAR_TIMEOUT },
+	{ "cellular_service", required_argument, 0, LOCATION_SHELL_OPT_CELLULAR_SERVICE },
+	{ "wifi_timeout", required_argument, 0, LOCATION_SHELL_OPT_WIFI_TIMEOUT },
+	{ "wifi_service", required_argument, 0, LOCATION_SHELL_OPT_WIFI_SERVICE },
+	{ 0, 0, 0, 0 }
+};
+
+/******************************************************************************/
+
+static void location_shell_print_usage(const struct shell *shell,
+				       struct location_shell_cmd_args *loc_cmd_args)
+{
+	switch (loc_cmd_args->command) {
+	case LOCATION_CMD_GET:
+		mosh_print_no_format(location_get_usage_str);
+		break;
+
+	default:
+		mosh_print_no_format(location_usage_str);
+		break;
+	}
+}
+
+#define MOSH_LOC_SERVICE_NONE 0xFF
+
+static enum location_service location_shell_string_to_service(const char *service_str)
+{
+	enum location_service service = MOSH_LOC_SERVICE_NONE;
+
+	if (strcmp(service_str, "any") == 0) {
+		service = LOCATION_SERVICE_ANY;
+	} else if (strcmp(service_str, "nrf") == 0) {
+		service = LOCATION_SERVICE_NRF_CLOUD;
+	} else if (strcmp(service_str, "here") == 0) {
+		service = LOCATION_SERVICE_HERE;
+	} else if (strcmp(service_str, "skyhook") == 0) {
+		service = LOCATION_SERVICE_SKYHOOK;
+	} else if (strcmp(service_str, "polte") == 0) {
+		service = LOCATION_SERVICE_POLTE;
+	}
+
+	return service;
+}
+
+/******************************************************************************/
+
+void location_ctrl_event_handler(const struct location_event_data *event_data)
+{
+	char gmaps_str[512];
```

Comment:
I think for example 128 bytes would be plenty for this, it would save some stack.

Reply:
https://github.com/nrfconnect/sdk-nrf/pull/6086/files#r751013691 will handle this comment.